Homework-8/script.js

```diff
@@ -0,0 +1,211 @@
+// Task 1
+
+// const makeCounter = () => {
+//   let count = 0;
+
+//   return () => {
+//     return count++;
+//   };
+// };
+
+// // independent counters
+
+// const counter = makeCounter();
+// const counter2 = makeCounter();
+
+// console.log(counter()); // 0
+// console.log(counter()); // 1
+
+// console.log(counter2()); // 0
+// console.log(counter2()); // 1
+
+// Task 2
+
+// function Counter() {
+//   let count = 0;
+
+//   this.up = function() {
+//     return ++count;
+//   };
+//   this.down = function() {
+//     return --count;
+//   };
+// }
+
+// const counter = new Counter();
+
+// console.log(counter.up()); // 1
+// console.log(counter.up()); // 2
+// console.log(counter.down()); // 1
+
+// Task 3
+
+// const phrase = "Hello";
+
+// if (true) {
+//   const user = "John";
+
+//   function sayHi() {
+//     console.log(`${phrase}, ${user}`);
+//   }
+// }
+
+// sayHi(); // Hello, John
+
+// Task 4
+
+// const sum = num1 => {
+//   return (num2) => {
+//       return num1 + num2;
+//   }
+// };
+
+// console.log(sum(2)(2));
+// console.log(sum(-5)(20));
+
+// Task 5
+
+// const arr = [1, 2, 3, 4, 5, 6, 7];
+
+// const inBetween = (from, to) => {
+//   return function() {
+//     if (arguments[0] >= from && arguments[0] <= to) {
+//       console.log(arguments[0]);
+//       return true;
+//     }
+//     return false;
+//   };
+// };
+
+// const inArray = arr => {
+//   return function() {
+//     return arr.includes(arguments[0]);
+//   };
+// };
+
+// console.log(arr.filter(inBetween(3, 6)));
+// console.log(arr.filter(inArray([1, 2, 10])));
+
+// Task 6
+
+// const users = [
+//   { name: "John", age: 20, surname: "Johnson" },
+//   { name: "Pete", age: 18, surname: "Peterson" },
+//   { name: "Ann", age: 19, surname: "Hathaway" }
+// ];
+
+// const byField = param => {
+//   return (a, b) => {
+//     return a[param] > b[param] ? 1 : -1;
```

Comment:
It can be confusing because of a and b usually are a numbers
personA[field] is much better